src/python_testing/TC_IDM_4_2.py

```diff
@@ -376,153 +474,188 @@ async def test_TC_IDM_4_2(self):
+        node_label_update_cb.wait_for_report()
+
+        # Save the time that the report is received
+        t_update_sec = time.time()
+
+        # Elapsed time between attribute subscription and write update
+        t_elapsed_sec = t_update_sec - t_report_sec
 
-        # Verify new attribute value after MinIntervalFloor time
-        asserts.assert_equal(new_node_label_read, new_node_label_write, "Attribute value not updated after write operation.")
+        # Verify that t_update - t_report is greater than min_interval_floor_s and less than the ReadClient SubscriptionTimeout
+        asserts.assert_greater(t_elapsed_sec, min_interval_floor_sec,
+                               f"t_update_sec - t_report_sec ({t_elapsed_sec}s) must be greater than min_interval_floor_sec ({min_interval_floor_sec}s)")
+        asserts.assert_less(t_elapsed_sec, subscription_timeout_sec,
+                            f"t_update_sec  - t_report_sec ({t_elapsed_sec}s) must be less than subscription_timeout_sec ({subscription_timeout_sec}s)")
 
         sub_cr1_update_value.Shutdown()
 
-        # *** Step 9 ***
-        self.print_step(
-            9, "CR1 sends a subscription request action for an attribute and set the MinIntervalFloor value to be greater than MaxIntervalCeiling.")
+        # *** Step 10 ***
+        # CR1 sends a subscription request action for an attribute and set the MinIntervalFloor
+        # value to be greater than MaxIntervalCeiling.
+        self.step(10)
 
-        # Subscribe to attribute with invalid reportInterval arguments, expect and exception
+        # Subscribe to attribute with invalid reportInterval arguments, expect an error
         sub_cr1_invalid_intervals = None
-        try:
+        with asserts.assert_raises(ChipStackError, "Expected exception wasn't thrown."):
             sub_cr1_invalid_intervals = await CR1.ReadAttribute(
                 nodeid=self.dut_node_id,
                 attributes=node_label_attr_path,
                 reportInterval=(20, 10),
                 keepSubscriptions=False
             )
-        except ChipStackError:
-            # Verify no subscription is established
-            with asserts.assert_raises(AttributeError):
-                sub_cr1_invalid_intervals.subscriptionId
-        except Exception:
-            asserts.fail("Expected exception was not thrown")
 
-        # *** Step 10 ***
-        self.print_step(
-            10, "CR1 sends a subscription request to subscribe to a specific global attribute from all clusters on all endpoints.")
+        # Verify no subscription was established
+        with asserts.assert_raises(AttributeError):
+            sub_cr1_invalid_intervals.subscriptionId
+
+        # *** Step 11 ***
+        # CR1 sends a subscription request to subscribe to a specific global attribute from
+        # all clusters on all endpoints.
+        self.step(11)
+
+        # Setting reportInterval values for steps 11-13
```

Comment:
why?

Reply:
For steps 3-8 and 11-13 the reportInterval values are not part of the testing per se, they just need to be there. The other steps update those values, so I'm just resetting them to "normal".

Reply:
as above - please leave min as-is.

Reply:
Updated to only set the max interval